bson/decoder_example_test.go

```diff
@@ -126,3 +128,85 @@ func ExampleDecoder_UseJSONStructTags() {
 	fmt.Printf("%+v\n", res)
 	// Output: {Name:Cereal Rounds SKU:AB12345 Price:399}
 }
+
+func ExampleDecoder_extendedJSON() {
+	// Define an Extended JSON document that contains the name, SKU, and price
+	// (in cents) of a product.
+	data := []byte(`{"name":"Cereal Rounds","sku":"AB12345","price_cents":{"$numberLong":"399"}}`)
+
+	// Create a Decoder that reads the Extended JSON document and use it to
+	// unmarshal the document into a Product struct.
+	vr, err := bsonrw.NewExtJSONValueReader(bytes.NewReader(data), true)
+	if err != nil {
+		panic(err)
+	}
+	decoder, err := bson.NewDecoder(vr)
+	if err != nil {
+		panic(err)
+	}
+
+	type Product struct {
+		Name  string `bson:"name"`
+		SKU   string `bson:"sku"`
+		Price int64  `bson:"price_cents"`
+	}
+
+	var res Product
+	err = decoder.Decode(&res)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("%+v\n", res)
+	// Output: {Name:Cereal Rounds SKU:AB12345 Price:399}
+}
+
+func ExampleDecoder_multipleExtendedJSONDocuments() {
+	// Define a newline-separated sequence of Extended JSON documents that
+	// contain X,Y coordinates.
+	data := []byte(`
+	{"x":{"$numberInt":"0"},"y":{"$numberInt":"0"}}
+	{"x":{"$numberInt":"1"},"y":{"$numberInt":"1"}}
+	{"x":{"$numberInt":"2"},"y":{"$numberInt":"2"}}
+	{"x":{"$numberInt":"3"},"y":{"$numberInt":"3"}}
+	{"x":{"$numberInt":"4"},"y":{"$numberInt":"4"}}
+	`)
```

Comment:
Even though this works, it is fairly common to align the string representation of a byte slice without indentation. For example: https://cs.opensource.google/go/go/+/master:src/encoding/json/fuzz_test.go;l=14

```suggestion
	data := []byte(`
{"x":{"$numberInt":"0"},"y":{"$numberInt":"0"}}
{"x":{"$numberInt":"1"},"y":{"$numberInt":"1"}}
{"x":{"$numberInt":"2"},"y":{"$numberInt":"2"}}
{"x":{"$numberInt":"3"},"y":{"$numberInt":"3"}}
{"x":{"$numberInt":"4"},"y":{"$numberInt":"4"}}
`)
```